plugins/modules/vmware_guest_disk.py

```diff
@@ -338,24 +378,67 @@ def create_scsi_controller(self, scsi_type, scsi_bus_number):
         scsi_ctl = vim.vm.device.VirtualDeviceSpec()
         scsi_ctl.operation = vim.vm.device.VirtualDeviceSpec.Operation.add
         scsi_ctl.device = self.scsi_device_type[scsi_type]()
-        scsi_ctl.device.unitNumber = 3
+        scsi_ctl.device.deviceInfo = vim.Description()
         scsi_ctl.device.busNumber = scsi_bus_number
-        scsi_ctl.device.hotAddRemove = True
         scsi_ctl.device.sharedBus = 'noSharing'
+        scsi_ctl.device.hotAddRemove = True
+        scsi_ctl.device.key = -randint(1000, 9999)
         scsi_ctl.device.scsiCtlrUnitNumber = 7
 
         return scsi_ctl
 
+    def create_sata_controller(self, sata_bus_number):
+        sata_ctl = vim.vm.device.VirtualDeviceSpec()
+        sata_ctl.operation = vim.vm.device.VirtualDeviceSpec.Operation.add
+        sata_ctl.device = vim.vm.device.VirtualAHCIController()
+        sata_ctl.device.deviceInfo = vim.Description()
+        sata_ctl.device.busNumber = sata_bus_number
+        sata_ctl.device.key = -randint(15000, 19999)
+
+        return sata_ctl
+
+    def create_nvme_controller(self, nvme_bus_number):
+        nvme_ctl = vim.vm.device.VirtualDeviceSpec()
+        nvme_ctl.operation = vim.vm.device.VirtualDeviceSpec.Operation.add
+        nvme_ctl.device = vim.vm.device.VirtualNVMEController()
+        nvme_ctl.device.deviceInfo = vim.Description()
+        nvme_ctl.device.busNumber = nvme_bus_number
+        nvme_ctl.device.key = -randint(31000, 39999)
+
+        return nvme_ctl
+
+    def find_disk_by_key(self, disk_key, disk_unit_number):
+        found_disk = None
+        for device in self.vm.config.hardware.device:
+            if isinstance(device, vim.vm.device.VirtualDisk) and device.key == disk_key:
+                if device.unitNumber == disk_unit_number:
+                    found_disk = device
+                    break
+
+        return found_disk
+
     @staticmethod
-    def create_scsi_disk(scsi_ctl_key, disk_index, disk_mode, disk_filename, sharing):
+    def get_parent_datacenter(obj):
```

Comment:
Please use the existing function from `module_utils/vmware.py`

Reply:
Sure, thanks.